CIP-0XXX/README.md

```diff
@@ -0,0 +1,561 @@
+   
+```
+
+#### Protocol messages
+
+* **MsgSubmitMessage(message)**: The client submits a message.
+* **MsgAcceptMessage**: The server accepts the message.
+* **MsgRejectMessage(reason)**: The server rejects the message and replies with a *reason*.
+* **MsgDone**: The client terminates the mini-protocol.
+
+#### Transition table
+
+| From state | Message | Parameters | To State|
+|--------|------|----|----| 
+| StIdle | MsgSubmitMessage | message | StBusy |
+| StBusy | MsgAcceptMessage |  | StIdle |
+| StBusy | MsgRejectMessage | reason | StIdle |
+| StIdle | MsgDone |  | StDone |
+
+
+## Local Message Notification mini-protocol
+
+### Description
+
+The local message notification mini-protocol is used by local clients to be notified about new message received by the core node. 
+
+The protocol follows a simple request-response pattern:
+
+1. The client sends a request with a single message.
+2. The server either accepts the message (and returns a confirmation) or rejects it (and returns the reason)
+
+### State machine
+
+| Agency ||
+|--------|------| 
+| Client has Agency | StIdle | 
+| Server has Agency | StBusy |
+
+```mermaid
+stateDiagram-v2   
+
+  classDef White fill:white,stroke:white
+  classDef Black fill:white,stroke:black
+  classDef Blue fill:white,stroke:blue
+  classDef Green fill:white,stroke:green
+
+  start:::White --> StIdle:::Blue;
+  StIdle:::Blue --> StBusy:::Green : MsgNextMessage
+  StBusy:::Green --> StIdle:::Blue : MsgHasMessage
+  StBusy:::Green --> StIdle:::Blue : MsgTimeoutMessage
+  StIdle:::Blue --> StDone:::Black : MsgDone
+   
+```
+
+#### Protocol messages
+
+* **MsgNextMessage**: The client asks for the next message.
+* **MsgHasMessage(message)**: The server has received a message.
+* **MsgTimeoutMessage**: The server has not received a message and times out.
+* **MsgDone**: The client terminates the mini-protocol.
+
+#### Transition table
+
+| From state | Message | Parameters | To State|
+|--------|------|----|----| 
+| StIdle | MsgNextMessage |  | StBusy |
+| StBusy | MsgHasMessage | message | StIdle |
+| StBusy | MsgTimeoutMessage | reason | StIdle |
+| StIdle | MsgDone |  | StDone |
+
+# Rationale
+
+## Why are we proposing this CIP?
+
+### For Mithril
+
+* Mithril requires strong network foundations to support interactions between its various nodes:
+
+  * Mithril needs to exist in a decentralized context where multiple aggregators can operate seamlessly and independantly.
+  * Mithril needs participation of all or nearly all of the Cardano network SPOs to provide maximal security to the multi-signatures embedded in the certificates.
+  * Creating a separate network would entail significant costs and efforts (there are more than 3,000 SPOs which would need to be connected with resilient and secure network, and much more passive nodes).
+  * The Cardano network is very efficient for diffusion (e.g. broadcasting) which is precisely what is needed for Mithril.
```

Comment:
Related to this or one bullet above:

Cardano SPOs need to be vigilant about what other applications run on their block producers and relay nodes. While a separate p2p network could be created, incoming connections must be treated carefully and all the same DoS considerations as with Cardano would need to apply. By standardizing the message diffusion of Mithril in the same way as the Cardano protocol stack, the additional risk of operating Mithril is greatly reduced.

Reply:
I added a new bullet point for this.